build_macos

```diff
@@ -1,14 +1,132 @@
 #!/usr/bin/env bash
 
-mkdir -p out
-
-g++ -fPIC src/*.cc -Wall -O0 \
--I include \
--I lib/gl3w/include \
-lib/gl3w/src/gl3w.c \
--I lib/glfw/include \
--L lib/glfw/src \
--lglfw \
--framework OpenGL \
--lpthread -ldl -lm \
--shared -o out/libburner.so
+BUILD_MODE=""
+
+if [ -z ${1} ]; then
+    echo "Compiling in development mode";
+    BUILD_MODE="DEVELOPMENT_MODE"
```

Comment:
I've seen `DEBUG` and `RELEASE` most often too. I that since DEVELOPMENT/DEBUG mode enables hot reloading, which can be viewed as both a development and a debug feature, I went with the more "positive" activity. (Not debugging something annoying, but playing around with the code.) I agree though. It probably makes sense to use `DEBUG`.